base/rational.jl

```diff
@@ -533,3 +533,13 @@ function hash(x::Rational{<:BitInteger64}, h::UInt)
     h = hash_integer(num, h)
     return h
 end
+
+# This method is only needed for performance. Since `first(r)` and `last(r)` have the same
+# denominator (because their difference is an integer), `length(r)` can be calulated without
+# calling `gcd`.
+function length(r::AbstractUnitRange{T}) where T<:Rational
```

Comment:
This method not only adds performance but changes behavior as well: Without this method, `length(::AbstractUnitRange{<:Rational})` uses `Rational` arithmetic, which is checked. The specialized method uses integer arithmetic, so it can overflow. 

I realize that we are okay with an overflowing `length`, but should this also apply to `Rational`s, which are using checked arithmetic by default? Or should the behavior be changed so that it matches `Rational` arithmetic?

 I will add a specialized `checked_length(::AbstractUnitRange{<:Rational})` method in any case.

Reply:
I think it is okay, since it should generally only overflow for extreme values that are unlikely